testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -1500,4 +1500,71 @@ public boolean isSupportedInTrino()
         CREATE_TABLE_AS_SELECT,
         CREATE_TABLE_WITH_NO_DATA_AND_INSERT,
     }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "storageFormats")
+    public void testSparkReadsTrinoTableAfterCleaningUp(StorageFormat storageFormat)
+    {
+        String baseTableName = "test_spark_reads_trino_partitioned_table_after_vacuum" + storageFormat;
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + trinoTableName);
+
+        onTrino().executeQuery(format("CREATE TABLE %s (_string VARCHAR, _bigint BIGINT) WITH (partitioning = ARRAY['_string'], format = '%s')", trinoTableName, storageFormat));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a', 1001)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a', 1002)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a', 1003)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('b', 1004)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('b', 1005)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('b', 1006)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('c', 1007)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('c', 1008)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('c', 1009)", trinoTableName));
+        onTrino().executeQuery((format("DELETE FROM %s WHERE _string = '%s'", trinoTableName, 'b')));
+
+        onTrino().executeQuery("SET SESSION iceberg.expire_snapshots_min_retention = '0s'");
+        onTrino().executeQuery(format("ALTER TABLE %s EXECUTE EXPIRE_SNAPSHOTS (retention_threshold => '0s')", trinoTableName));
+
+        Row row1 = row(3006);
```

Comment:
can we add single line sanity check that `expire_snapshots` actually did something to table structure here. So we know we are not testing "spark can read data from t after no-op on Trino".